### PR TITLE
'Reset all' in 'Filter' not working, closes #676

### LINKED ISSUE #676: 'Reset all' in 'Filter' not working
Steps:  

1.Visit https://apinf.io

2.Sign in

3.Select 'API Backends' from menu

4.Click on 'Add API Backends'

5.Go to 'Manage API backends'

6.Click on 'View' button

7.Click on 'View usage instructions'

8.Click on the link for 'profile page' and copy the API key

9.Click on the link given for making API call   

10.Add the Frontend prefix and API key

11.Select 'Analytics' from menu    
12.Go to Overview chart and select few records  
13. Click on 'Reset All'  

Result:  'Reset all'  is not working

Screenshot:  

reset_all_not_working2

### PR DESCRIPTION
**Pull request by jykae**
WIP

### CODE REVIEW

**Comment by jykae** [comment]
@apinf/developers please review

File: client/views/dashboard/charts/filter/filter.html [hunk 1]
```diff
@@ -7,7 +7,7 @@ <h3 class="box-title">{{_ "filterData_Title"}}</h3>
     <div class="box-body">
       <form id="filteringForm">
         {{_ "filterData_Month"}}
-        <select name="month" >
+        <select id="month" name="month" >
```
Comment (brylie) [inline client/views/dashboard/charts/filter/filter.html]: Perhaps prefix these IDs with `filter-`, to be a bit more specific. E.g.

* `filter-month`
* `filter-year`
* `filter-limit`
* `filter-reset`

File: client/views/dashboard/charts/filter/filter.js [hunk 2]
```diff
@@ -36,5 +36,41 @@ Template.chartsLayout.events({
 
     instance.getDashboardData(input)
 
+  },
+  "click #reset": function(event, template){
+    // appending loading state
+    $('#loadingState').html("Loading...");
+
```
Comment (brylie) [inline client/views/dashboard/charts/filter/filter.js]: Cleanup whitespace.

File: client/views/dashboard/charts/filter/filter.js [hunk 3]
```diff
@@ -36,5 +36,41 @@ Template.chartsLayout.events({
...
+
+    // gets current month and year -> providing them for initial query
+    var currentYearAndMonth = moment().format("YYYY-MM");
+
+    // sets default items amount to be returned
+    var initialLimit = 10000;
+
+    // Initialize filter values
+    $('#month option[value="00"]').prop('selected', true);
```
Comment (brylie) [inline client/views/dashboard/charts/filter/filter.js]: Try something simpler here. E.g.

```js
$('#month-select').val('1');
```

**Comment by jykae** [comment]
Requested improvements done.

**Pull request merged**